eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/client/EclairRpcClient.scala

```diff
@@ -442,25 +458,49 @@ class EclairRpcClient(val instance: EclairInstance)(
       channelId: ChannelId,
       feeBaseMsat: MilliSatoshis,
       feeProportionalMillionths: Long): Future[Unit] = {
-    eclairCall[Unit]("updaterelayfee",
-                     List(JsString(channelId.hex),
-                          JsNumber(feeBaseMsat.toLong),
-                          JsNumber(feeProportionalMillionths)))
+    eclairCallNew[Unit]("updaterelayfee",
+      "channelId" -> channelId.hex,
+      "feeBaseMsat" -> feeBaseMsat.toLong.toString,
+      "feeProportionalMillionths" -> feeProportionalMillionths.toString)
   }
 
   override def updateRelayFee(
       shortChannelId: ShortChannelId,
       feeBaseMsat: MilliSatoshis,
-      feePropertionalMillionths: Long): Future[Unit] = {
-    eclairCall[Unit]("updaterelayfee",
-                     List(JsString(shortChannelId.hex),
-                          JsNumber(feeBaseMsat.toLong),
-                          JsNumber(feePropertionalMillionths)))
+      feeProportionalMillionths: Long): Future[Unit] = {
+    eclairCallNew[Unit]("updaterelayfee",
+      "shortChannelId" -> shortChannelId.toHumanReadableString,
+      "feeBaseMsat" -> feeBaseMsat.toLong.toString,
+      "feeProportionalMillionths" -> feeProportionalMillionths.toString)
   }
 
   // TODO: channelstats, audit, networkfees?
   // TODO: Add types
 
+  private def eclairCallNew[T](command: String, parameters: (String, String)*)(
```

Comment:
So one thing we could do here to make this a bit safer, is replace the type signature here and add an implicit conversion: 

```scala
 import scala.language.implicitConversions
  implicit private def obj2Json[T](obj: T)(
      implicit writes: Writes[T]): JsValue = Json.toJson(obj)

  private def eclairCallNew[T](command: String, parameters: (String, JsValue)*)
```

Then in the rest of the code, instead of passing in strings that are serialized ad hoc we could use all the JSON serializers we have defined already in Bitcoin-S. 

If you add this import at the top:

```
import org.bitcoins.rpc.serializers.JsonSerializers._
```

You get serialization to and from JSON for most data structures for free:-)

Reply:
New eclair API doesn't use JSON for encoding API parameters anymore, it uses plain HTTP forms instead.

Reply:
Hm. Still, we'd like to avoid ad-hoc serialization of things in each RPC call. For example if doing `.toString` on an address, I'm pretty sure it'll look like `P2SHBitcoindAddress(actualAddressHere)` instead of just `actualAddressHere`.

The JSON serializers we have already handle correct conversion of data structures to string representation, so it'd be nice to use those somehow

Reply:
In order to create an HTTP form in akka-http I need a Seq[(String, String)]. Do you want me to create Seq[(String, JValue]) using the implictis and than to convert the seq into Seq[(String, String)] somehow?

Reply:
I think that's a good idea. For example here, you can see the JSON writer for addresses: https://github.com/bitcoin-s/bitcoin-s/blob/839d520206023fa1075a65e19392d9f9b72e3a62/bitcoind-rpc/src/main/scala/org/bitcoins/rpc/serializers/JsonWriters.scala#L42-L44

As you can see, its calling `.value`, instead of `.toString`. IMO its cleaner to have string serialization happen in one place (the JSON writers) instead of ad hoc in each RPC call.

But again, feel free to disagree on this! I'm open to it being a bad idea:-)

Reply:
There's a test case to make sure that the addresses get serialized in the way Eclair can parse them.